eternalcore-core/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatCommands.java

```diff
@@ -49,13 +50,25 @@ void socialSpy(Player player) {
         UUID uuid = player.getUniqueId();
 
         if (this.privateChatService.isSpy(uuid)) {
-            this.privateChatService.disableSpy(uuid);
-            this.noticeService.player(uuid, translation -> translation.privateChat().socialSpyDisable());
+            this.modifySocialSpy(uuid, this.privateChatService::disableSpy);
             return;
         }
 
-        this.noticeService.player(uuid, translation -> translation.privateChat().socialSpyEnable());
-        this.privateChatService.enableSpy(uuid);
+        this.modifySocialSpy(uuid, this.privateChatService::enableSpy);
+    }
+
+    private void modifySocialSpy(UUID uuid, Consumer<UUID> featureModifier) {
```

Comment:
Consumer to trochę przesada, wystarczy zwykły call metody. Przy tym zmień nazwę metody tak aby mówiła, że wysyła wiadomość.